packages/core/src/errors/error-reporter.ts

```diff
@@ -66,11 +73,25 @@ export class ErrorReporter {
 		release,
 		environment,
 		serverName,
+		releaseDate,
 	}: ErrorReporterInitOptions) {
 		process.on('uncaughtException', (error) => {
 			this.error(error);
 		});
 
+		if (releaseDate) {
+			const releaseExpiresInMs = releaseDate.getTime() + SIX_WEEKS_IN_MS - Date.now();
+			if (releaseExpiresInMs <= 0) {
+				this.logger.warn(RELEASE_EXPIRATION_WARNING);
+				return;
+			}
+			// Once this release expires, reject all events
+			setTimeout(() => {
```

Comment:
This timer will prevent the instance from shutting down gracefully. We need to clear it in `shutdown`